### PR TITLE
Contributing and Readme

### PR DESCRIPTION
**Pull request by zeroset**
Cleanup and update README and Contributing

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -25,35 +25,68 @@ There are only a few rules and advices we want to give to you:
...
-
 ## Advice for new contributors
 
-### First steps
-* Clone/fork our github/gitlab repository
-~~~
-git clone https://cst.version.fz-juelich.de/jupedsim/jpscore.git
-~~~
+### Building from source
+The build is tested on Ubuntu 18.04 with gcc-8 / clang-8 and on macOS Mojave
```
Comment (chraibi) [inline CONTRIBUTING.md]: macOS Catalina

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -25,35 +25,68 @@ There are only a few rules and advices we want to give to you:
...
+10.14.5 and apple clang.
 
-* Change to the developement branch and create a branch with your feature.
-~~~
-git checkout developement 
-git checkout -b feature_name
-~~~
+#### Requirements
+For the compilation of JuPedSim core you need a C++17 capable compiler and standard library.
```
Comment (chraibi) [inline CONTRIBUTING.md]: what is a "standard librarary"? How can it be installed?

File: CONTRIBUTING.md [hunk 3]
```diff
@@ -25,35 +25,68 @@ There are only a few rules and advices we want to give to you:
...
+Required:
+* boost (>= 1.65)
+* libomp (if you want to use OpenMP with Apple LLVm or Clang on Linux)
+* cmake (>= 3.1)
 
-* Assuming you are in the jpscore folder type 
+Recommended:
+* ninja-build
+
+The following dependencies could be installed system wide or will be installed locally using a script:
```
Comment (chraibi) [inline CONTRIBUTING.md]: using the install [script](https://github.com/JuPedSim/jpscore/blob/develop/scripts/setup-deps.sh)

File: CONTRIBUTING.md [hunk 4]
```diff
@@ -25,35 +25,68 @@ There are only a few rules and advices we want to give to you:
...
+* spdlog (>= 1.3)
+* fmtlib (>= 6.0)
+* CLI11 (>= 1.8)
+* catch2
+
+#### How to build
+Once you have installed the dependencies continue building as follows:
```
Comment (chraibi) [inline CONTRIBUTING.md]: how?

This step comes later...

File: CONTRIBUTING.md [hunk 5]
```diff
@@ -25,35 +25,68 @@ There are only a few rules and advices we want to give to you:
...
+```bash
+git clone https://github.com/JuPedSim/jpscore.git
```
Comment (chraibi) [inline CONTRIBUTING.md]: maybe with some limited depth? e.g. `--depth 5`

File: CONTRIBUTING.md [hunk 6]
```diff
@@ -25,35 +25,68 @@ There are only a few rules and advices we want to give to you:
...
+cd jpscore
+mkdir build && cd build
+./../scripts/setup-deps.sh
+cmake -DCMAKE_PREFIX_PATH=$(pwd)/deps ..
+make -j$(nproc)
```
Comment (chraibi) [inline CONTRIBUTING.md]: but `ninja` is preferred, right?

File: CONTRIBUTING.md [hunk 7]
```diff
@@ -25,35 +25,68 @@ There are only a few rules and advices we want to give to you:
...
 ```
-mkdir build && cd build 
-cmake .. 
+
+Note: If you do not want to use OpenMP you have to pass `-DUSE_OPENMP=OFF` to
```
Comment (chraibi) [inline CONTRIBUTING.md]: I would write this note right after the first cmake call

File: CONTRIBUTING.md [hunk 8]
```diff
@@ -25,35 +25,68 @@ There are only a few rules and advices we want to give to you:
...
+cmake on generation.
+```bash
+cmake -DUSE_OPENMP=OFF -DCMAKE_PREFIX_PATH=$(pwd)/deps ..
```
Comment (chraibi) [inline CONTRIBUTING.md]: probably this is not necessary since the cmake flags will follow

File: CONTRIBUTING.md [hunk 9]
```diff
@@ -25,35 +25,68 @@ There are only a few rules and advices we want to give to you:
...
+##### BUILD_TESTING defaults to OFF
+Build full system tests and add them to ctest
+
+##### BUILD_WITH_ASAN defaults to OFF (Does not support Windows)
+Build an additional target `jpscore_asan` with address and undefined behavior
+sanitizer enabled. Note there is an approx. 2x slowdown when using
+`jpscore_asan` over `jpscore`
 
 ### Workflow
 The branches **master** and **develop** are **protected**. You can **only push to your feature-branch.**
```
Comment (chraibi) [inline CONTRIBUTING.md]: this is not true. 
I think now all goes with PR..

File: CONTRIBUTING.md [hunk 10]
```diff
@@ -64,15 +97,6 @@ After doing that open a merge/pull request.
 
 If your fix/feature is accepted it will be merged into the develop-branch.
```
Comment (chraibi) [inline CONTRIBUTING.md]: I would say that PR is the way to go..

File: README.md [hunk 11]
```diff
@@ -10,48 +10,44 @@ The core module for performing simulations. See the [online documentation](http:
 ## Building from source
 
 ### Linux / macOS
-The build is tested on Ubuntu 18.04 with gcc-9.1 / clang-8 and on macOS Mojave
-10.14.5 with dependencies from Homebrew and Apple LLVM 10.
+The build is tested on Ubuntu 18.04 with gcc-8 / clang-8 and on macOS Mojave
```
Comment (chraibi) [inline README.md]: macOS Catarina

File: README.md [hunk 12]
```diff
@@ -10,48 +10,44 @@ The core module for performing simulations. See the [online documentation](http:
...
+10.14.5 and apple clang.
```
Comment (chraibi) [inline README.md]: `AppleClang`

File: README.md [hunk 13]
```diff
@@ -10,48 +10,44 @@ The core module for performing simulations. See the [online documentation](http:
...
 
 #### Requirements
-On Linux you will need a C++17 capable compiler and a standard library that
-supports `<filesystem>`. On macOS you will need the system compiler with C++17
-and filesystem from boost.
+For the compilation of JuPedSim core you need a C++17 capable compiler and standard library.
```
Comment (chraibi) [inline README.md]: For the compilation of `jpscore` you need a C++17 capable compiler and standard library.

**Comment by chraibi** [comment]
It seems that constructions in Contributing and readme are duplicated

File: CONTRIBUTING.md (hunk 1 above)
Comment (schroedtert) [reply CONTRIBUTING.md]: Travis still uses Mojave (macOS 10.14)

**Pull request closed without merging**